lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
Then do the teardown before the join?

I agree that we probably can't solve this for everyone, but it seems like we'll have a list of a bunch of threads that we can join after whatever teardown is done. If the join takes too long, then we fail the test with an explanation on how they can make sure that the test doesn't fail (provide a free method? register a resource?).